### PR TITLE
Merge upstream2

### PR DESCRIPTION
**Pull request by hexadecimalDinosaur**

### CODE REVIEW

File: judge/admin/profile.py [hunk 1]
```diff
@@ -54,18 +54,20 @@ def has_add_permission(self, request):
...
+                                          'username_display_override')}),
         (_('Text Fields'), {'fields': ('about', 'user_script')}),
     )
-    list_display = ('user', 'full_name', 'email', 'is_totp_enabled', 'is_external_user',
+    readonly_fields = ('user',)
+    list_display = ('admin_user_admin', 'email', 'is_totp_enabled', 'timezone_full',
+                    'full_name', 'is_external_user',
                     'date_joined', 'last_access', 'ip', 'show_public')
     ordering = ('user__username',)
-    search_fields = ('user__username', 'user__first_name', 'user__last_name', 'ip', 'user__email')
```
Comment (fluix-dev) [inline judge/admin/profile.py:68]: Any reason to remove these?

File: judge/models/contest.py [hunk 2]
```diff
@@ -134,11 +134,19 @@ class Contest(models.Model):
...
     run_pretests_only = models.BooleanField(verbose_name=_('run pretests only'),
                                             help_text=_('Whether judges should grade pretests only, versus all '
                                                         'testcases. Commonly set during a contest, then unset '
                                                         'prior to rejudging user submissions when the contest ends.'),
                                             default=False)
+    show_short_display = models.BooleanField(verbose_name=_('show short form settings display'),
+                                             help_text=_('Whether to show a section containing contest settings '
+                                                         'on the contest page or not.'),
+                                             default=False)
+    is_organization_private = models.BooleanField(verbose_name=_('private to organizations'), default=False)
```
Comment (fluix-dev) [inline judge/models/contest.py]: We already have this above. Its uses elsewhere should be looked at carefully to (though I think most are unit tested which is nice).

File: judge/models/profile.py [hunk 3]
```diff
@@ -140,6 +144,8 @@ class Profile(models.Model):
                                            help_text=_('If set, this account is an external user, and '
                                                        'has extra restrictions.'))
     data_last_downloaded = models.DateTimeField(verbose_name=_('last data download time'), null=True, blank=True)
+    username_display_override = models.CharField(max_length=100, blank=True, verbose_name=_('display name override'),
```
Comment (fluix-dev) [inline judge/models/profile.py:147]: Note: We may want to take advantage of this for LCC/Moose.

File: judge/views/contests.py [hunk 4]
```diff
@@ -595,16 +612,27 @@ def get_context_data(self, **kwargs):
...
-        if cached is not None:
-            return HttpResponse(cached)
-        response = super(CachedContestCalendar, self).render()
-        response.render()
-        cached.set(key, response.content)
-        return response
+class ContestICal(TitleMixin, ContestListMixin, BaseListView):
+    def generate_ical(self):
+        cal = ICalendar()
+        cal.add('prodid', '-//DMOJ//NONSGML Contests Calendar//')
```
Comment (fluix-dev) [inline judge/views/contests.py:618]: Hm, mind taking a look at the spec? I think this should be changed to "MCPT" but I'm not *quite* certain.

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Looks pretty good to me, just the comments and some questions below. Mainly asking them since this is one of the first few merges and want to make sure important changes were considered.

 - Should our custom contest formats have `get_short_form_display` defined?
 - Are there other places we should switch to `username_override` or `display_name`?
 - Should we be hardcore about 2FA?

Finally, once we decide to merge, being careful with the rating migration is likely warranted. I don't quite recall what the rating situation is on our end (do we just not display it?)

File: judge/views/contests.py (hunk 4 above)
Comment (nyiyui) [reply judge/views/contests.py:618]: `prodid` is the name of the generator[^1]. It should be either "WLMOJ" or "DMOJ". I tihnk "DMOJ" is fine as a) the iCal code has (basically) no changes from upstream and b) less diffs between upstream.
[^1]: https://www.kanzaki.com/docs/ical/prodid.html

File: judge/admin/profile.py (hunk 1 above)
Comment (nyiyui) [reply judge/admin/profile.py:68]: No specific reason other than that (I think) username display overrides were added and names don't make too much sense anymore (also I think users can't set them).

File: judge/models/contest.py (hunk 2 above)
Comment (nyiyui) [reply judge/models/contest.py]: The duplicate is removed. The other uses seem fine.

File: judge/admin/profile.py (hunk 1 above)
Comment (hexadecimalDinosaur) [reply judge/admin/profile.py:68]: Names are set by users on signup to the site (MCPT specific, not found on DMOJ), searching by name is useful for us.

File: judge/views/contests.py (hunk 4 above)
Comment (fluix-dev) [reply judge/views/contests.py:618]: Was the direction I was leaning as well. Let's keep `DMOJ`.

**Review by fluix-dev — APPROVED** [review]
LGTM.

**Pull request merged**